core/controllers/editor.py

```diff
@@ -774,6 +775,9 @@ def post(self, exploration_id):
 
         fs.commit(self.user_id, filepath, raw, mimetype=mimetype)
```

Comment:
I mentioned this before -- but why don't you create all three versions of the image at the same time, rather than create one here and create the other two elsewhere?

Reply:
The reason behind this is, we have our one_off job which is going to use FileModel so that counts for one version already. In order to have same function for both of them ( one off job and the editor) I have to create one version here in the editor and the other versions in the create_compressed_versions_of_image. Plus the name suggest that the compressed versions of the provided original image.

Reply:
One solution could be to add a check in the create function, which sees if the original image is already there in the datastore or not. If it exists then create two, else create all three.
Let me know what you think

Reply:
Oh, I see what you mean. OK, let's do this then:

- have exp_services.create_image_with_compressed_versions() which creates 3 images
- have a temporary exp_services.create_compressed_versions_of_images() which creates 2 images

After we do the migration, we delete the second one. If you think about this in the long term, only the first one should remain.

What do you think?

Reply:
Yeah done